### ISSUE TITLE
Multirun fails with Webstorm 2021.2

### ISSUE THREAD

**Issue by JoschuaGoetz** [rkhmelyuk/multirun#78]
Multirun crashes with the new version of Webstorm.
Error:
```
Error running 'Run'
EventQueue.isDispatchThread()=true Toolkit.getEventQueue()=com.intellij.ide.IdeEventQueue@623a6c4b
Current thread: Thread[AWT-EventQueue-0,6,main] 905569093
SystemEventQueueThread: Thread[AWT-EventQueue-0,6,main] 905569093

```

I haven't tested with other jetbrains IDEA's

**Comment by NicolaF** [comment]
Hi,

exact same issue on IntelliJ 2021.2. Here is the full stack trace:
```
2021-07-28 13:58:54,656 [  30775]   INFO - xecution.runners.ExecutionUtil - Error running '<REDACTED>':<br>EventQueue.isDispatchThread()=true Toolkit.getEventQueue()=com.intellij.ide.IdeEventQueue@33bb8a0f
Current thread: Thread[AWT-EventQueue-0,6,main] 1131281534
SystemEventQueueThread: Thread[AWT-EventQueue-0,6,main] 1131281534 
com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: EventQueue.isDispatchThread()=true Toolkit.getEventQueue()=com.intellij.ide.IdeEventQueue@33bb8a0f
Current thread: Thread[AWT-EventQueue-0,6,main] 1131281534
SystemEventQueueThread: Thread[AWT-EventQueue-0,6,main] 1131281534
	at com.intellij.openapi.application.impl.ApplicationImpl.throwThreadAccessException(ApplicationImpl.java:1030)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertIsNonDispatchThread(ApplicationImpl.java:1013)
	at com.intellij.execution.impl.RunManagerImpl$Companion.canRunConfiguration(RunManagerImpl.kt:86)
	at com.intellij.execution.impl.RunManagerImpl.canRunConfiguration(RunManagerImpl.kt)
	at com.khmelyuk.multirun.MultirunRunnerState.checkRunConfiguration(MultirunRunnerState.java:294)
	at com.khmelyuk.multirun.MultirunRunnerState.runConfigurations(MultirunRunnerState.java:91)
	at com.khmelyuk.multirun.MultirunRunnerState.runConfigurations(MultirunRunnerState.java:278)
	at com.khmelyuk.multirun.MultirunRunnerState.runConfigurations(MultirunRunnerState.java:278)
	at com.khmelyuk.multirun.MultirunRunnerState.runConfigurations(MultirunRunnerState.java:278)
	at com.khmelyuk.multirun.MultirunRunnerState.runConfigurations(MultirunRunnerState.java:278)
	at com.khmelyuk.multirun.MultirunRunnerState.runConfigurations(MultirunRunnerState.java:278)
	at com.khmelyuk.multirun.MultirunRunnerState.runConfigurations(MultirunRunnerState.java:278)
	at com.khmelyuk.multirun.MultirunRunnerState.runConfigurations(MultirunRunnerState.java:278)
	at com.khmelyuk.multirun.MultirunRunnerState.execute(MultirunRunnerState.java:67)
	at com.intellij.execution.runners.DefaultProgramRunnerKt.executeState(DefaultProgramRunner.kt:32)
	at com.intellij.execution.runners.DefaultProgramRunner.doExecute(DefaultProgramRunner.kt:26)
	at com.intellij.execution.runners.DefaultProgramRunner$execute$1.invoke(DefaultProgramRunner.kt:20)
	at com.intellij.execution.runners.DefaultProgramRunner$execute$1.invoke(DefaultProgramRunner.kt:14)
	at com.intellij.execution.impl.ExecutionManagerImpl$startRunProfile$1.invoke(ExecutionManagerImpl.kt:171)
	at com.intellij.execution.impl.ExecutionManagerImpl$startRunProfile$1.invoke(ExecutionManagerImpl.kt:70)
	at com.intellij.execution.impl.ExecutionManagerImpl$doStartRunProfile$startRunnable$1.run(ExecutionManagerImpl.kt:232)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:214)
	at com.intellij.openapi.application.TransactionGuardImpl.access$200(TransactionGuardImpl.java:21)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:196)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:794)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeLater$4(ApplicationImpl.java:350)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:84)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:133)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:46)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:189)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:886)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:755)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:442)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:825)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:441)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:794)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:487)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by rkhmelyuk** [comment]
Let me take a look to it over weekend to figure out what's happening there.

**Comment by richardbrockie** [comment]
Also on PyCharm 2021.2:

`2:42 PM	Error running 'OTD dev server + Celery'
				EventQueue.isDispatchThread()=true Toolkit.getEventQueue()=com.intellij.ide.IdeEventQueue@8ab428f
				Current thread: Thread[AWT-EventQueue-0,6,main] 1061497835
				SystemEventQueueThread: Thread[AWT-EventQueue-0,6,main] 1061497835`

@rkhmelyuk Thanks for looking into it - this plugin is really useful!

**Comment by oouo** [comment]
@rkhmelyuk Thanks for the hard work! The plugins are great!

**Comment by rkhmelyuk** [comment]
Hi Everyone, thanks for reporting this issue and providing stacktrace. Indeed there was a couple of issues with 2021.2 version of IDE. I have fixed them, and uploaded a new Multirun version - 1.13. 

I have done a few tests in IntellijIDEA and everything seems to be working properly. But let me know if you find some issues with this new version.

**Comment by richardbrockie** [comment]
> Hi Everyone, thanks for reporting this issue and providing stacktrace. Indeed there was a couple of issues with 2021.2 version of IDE. I have fixed them, and uploaded a new Multirun version - 1.13.
> 
> I have done a few tests in IntellijIDEA and everything seems to be working properly. But let me know if you find some issues with this new version.

@rkhmelyuk - confirmed fixed in PyCharm 2021.2 with 1.13 - thanks for such a quick fix!

**Comment by JoschuaGoetz** [comment]
> Hi Everyone, thanks for reporting this issue and providing stacktrace. Indeed there was a couple of issues with 2021.2 version of IDE. I have fixed them, and uploaded a new Multirun version - 1.13.
> 
> I have done a few tests in IntellijIDEA and everything seems to be working properly. But let me know if you find some issues with this new version.

@rkhmelyuk I could also verify it being fixed on Webstorm 2021.2.
Thanks for the quick fix and the useful extension.

I'll close the issue as i opened it. 
Feel free to reopen if you want it open for clarity sake
